src/orion/core/worker/primary_algo.py

```diff
@@ -348,9 +361,33 @@ def _verify_trial(self, trial: Trial, space: Optional[Space] = None) -> None:
             )
 
 
+def get_original_parent(registry, transformed_space, trial_parent_id):
```

Comment:
Type hints are sort-of required here given the context.
Also,  IMO any new function that gets added to a file without type hints should also have valid type hints.

```suggestion
def get_original_parent(registry: Registry, transformed_space: TransformedSpace, trial_parent_id: str) -> Trial:
```